python/cudf/cudf/core/column/lists.py

```diff
@@ -408,6 +409,22 @@ def contains(self, search_key: ScalarLike) -> ParentType:
         else:
             return res
 
+    def index(self, search_key: ScalarLike) -> ParentType:
+        search_key = cudf.Scalar(search_key)
+        try:
+            res = self._return_or_inplace(
+                index_of_scalar(self._column, search_key)
+            )
+        except RuntimeError as e:
+            if (
+                "Type/Scale of search key does not "
+                "match list column element type." in str(e)
+            ):
+                raise TypeError(str(e)) from e
```

Comment:
I just noticed that the corresponding exception testing in the `contains` method above is missing the space after "not" on the first line of the exception check, so it won't actually convert `RuntimeError` to `TypeError` as desired. Also no need to type out the full error message twice there, we could use the same strategy of `TypeError(str(e))` that's used here. It's slightly out of scope for this PR, but since it's a quick bug fix could you add it anyway? I can't make a suggestion because it's outside the diff.